components/com_apps/css/apps.css

```diff
@@ -8,4 +8,9 @@
 
 .row-fluid .extensions {
     width: 75%;
-}
\ No newline at end of file
+}
+
+.com-apps-container .thumbnail .item-image img {
+	max-width: 100%;
+	max-height: 100%;
```

Comment:
I think max-height should be around 90%. That fixes the issue for images that are narrow and tall.